utils/compute-perm-free-error.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python
+# encoding: utf-8
+
+# Copyright 2017 Johns Hopkins University (Shinji Watanabe)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import codecs
+import json
+import logging
+import numpy as np
+import re
+import six
+import sys
+
+from minpermwer import get_utt_permutation
+
+is_python2 = sys.version_info[0] == 2
+
+
+def get_results(ref, hyp, args):
+    result_key = 'r{}h{}'.format(ref, hyp)
+    result_file = getattr(args, 'result_' + result_key)
+
+    re_id = r'^id: '
+    re_strings = {'Speaker': r'^Speaker sentences',
+                  'Scores': r'^Scores: ',
+                  'REF': r'^REF: ',
+                  'HYP': r'^HYP: '}
+    re_id = re.compile(re_id)
+    re_patterns = {}
+    for p in re_strings.keys():
+        re_patterns[p] = re.compile(re_strings[p])
+
+    results = {}
+    tmp_id = None
+    tmp_ret = {}
+
+    with open(result_file, 'r') as f:
+        line = f.readline()
+        while line:
+            x = line.rstrip()
+            x_split = x.split()
+
+            if re_id.match(x):
+                if tmp_id:
+                    results[tmp_id] = {result_key: tmp_ret}
+                    tmp_ret = {}
+                tmp_id = x_split[1]
+            for p in re_patterns.keys():
+                if re_patterns[p].match(x):
+                    tmp_ret[p] = ' '.join(x_split[1:])
+            line = f.readline()
+
+    if tmp_ret != {}:
+        results[tmp_id] = {result_key: tmp_ret}
+
+    return result_key, {'utts': results}
+
+
+def merge_results(results):
+    rslt_lst = []
+
+    # make intersection set for utterance keys
+    intersec_keys = []
+    for x in results.keys():
+        j = results[x]
+
+        ks = j['utts'].keys()
+        logging.info(x + ': has ' + str(len(ks)) + ' utterances')
+
+        if len(intersec_keys) > 0:
+            intersec_keys = intersec_keys.intersection(set(ks))
+        else:
+            intersec_keys = set(ks)
+        rslt_lst.append(j)
+
+    logging.info('After merge, the result has ' + str(len(intersec_keys)) +
+                 ' utterances')
+
+    # merging results
+    dic = dict()
+    for k in intersec_keys:
+        v = rslt_lst[0]['utts'][k]
+        for j in rslt_lst[1:]:
+            v.update(j['utts'][k])
+        dic[k] = v
+
+    return dic
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--num-spkrs', type=int, default=2,
+                        help='number of mixed speakers.')
+    parser.add_argument('result_r1h1', type=str,
```

Comment:
Maybe you use `nargs='+'` and accept any numbers of files to generalize this scoring script for any speakers. Then, we can check that the number of files is consistent with the number of speakers later.